src/traincontroller-builder.lua

```diff
@@ -322,6 +322,11 @@ function Traincontroller.Builder:buildNextTrain(trainBuilderIndex)
                     name  = "trainassembly-trainfuel",
                     count = ingredient.amount,
                   }
+                elseif game.item_prototypes[ingredient.name].fuel_value > 0 then
```

Comment:
It is very performance heavy to index the `game.item_prototypes` table. It would be better if you cache this value in a separate table and only index the item prototype table when a new item is found.

I suggest creating a function `Trainassembly::isFuelItem(itemName)` which could index a cached table by the name. If the value (true/false/nil) is not equal to nil, it returns the cached value. If it is nil, it adds the value to the table and returns that value.

Reply:
Thank you. I've updated this with the caching added

Reply:
implemented in https://github.com/voske123/FactorioMod-trainConstructionSite/pull/90/commits/392af13135506549aa2d48db78fc3737f8c66fe0